tests/sync.rs

```diff
@@ -152,6 +152,38 @@ fn recv_before_send() {
     })
 }
 
+#[cfg(all(feature = "std", loom))]
+#[test]
+fn send_recv_different_threads() {
```

Comment:
These two tests are not perfect. But they do trigger an error on the commit where you pointed out that I handled the `UNPARKING` state invalidly, @Cassy343. And they pass now. So they at least protect against some regression.

Reply:
These look good to me!